### PR TITLE
Make decoder_test_helper.h accept TensorView

### PR DESCRIPTION
**Pull request by jantonguirao**
Signed-off-by: Joaquin Anton <janton@nvidia.com>

## Category:

**Other** 

## Description:

It enhances the decoder test utils so that the functions can work with TensorView (not only Tensor).

### Affected modules and functionalities:

Test utils 

### Key points relevant for the review:

All
### Tests:

- [x] Existing tests apply
- [ ] New tests added
  - [ ] Python tests
  - [ ] GTests
  - [ ] Benchmark
  - [ ] Other
- [ ] N/A

### Documentation
- [x] Existing documentation applies
- [ ] Documentation updated
  - [ ] Docstring
  - [ ] Doxygen
  - [ ] RST
  - [ ] Jupyter
  - [ ] Other
- [ ] N/A

#### Requirements
- [ ] Implements new requirements
- [ ] Affects existing requirements
- [x] N/A

**REQ IDs**: N/A

**JIRA TASK**: N/A

### CODE REVIEW

File: dali/imgcodec/decoders/decoder_test_helper.h [hunk 1]
```diff
@@ -134,30 +132,75 @@ class DecoderTestBase : public ::testing::Test {
...
     ), DALI_FAIL(make_string("Unsupported reference type: ", ref.type())));  // NOLINT
   }
 
+
   /**
   * @brief Crops a tensor to specified roi_shape, anchored at roi_begin.
   * Does not support padding.
   */
+  template <int ndim>
+  void Crop(const TensorView<StorageCPU, OutputType, ndim> &output,
```
Comment (szkarpinski) [inline dali/imgcodec/decoders/decoder_test_helper.h:189]: Sometimes we want to `Crop` the reference, which is not necessarily of type `OutputType` (when testing things such us color conversion it's convenient to keep the reference as float). What do you think about templating this `Crop` with image type as well?
```cpp
template <typename ImageType, int ndim>
void Crop(const TensorView<StorageCPU, ImageType, ndim> &output,
...
```

Comment (jantonguirao) [reply dali/imgcodec/decoders/decoder_test_helper.h]: done

**Review by szkarpinski — APPROVED** [review]

**Comment by jantonguirao** [comment]
!build

**Review by szalpal — APPROVED** [review]

**Comment by szkarpinski** [comment]
!build

**Review by stiepan — APPROVED** [review]

**Comment by szkarpinski** [comment]
!build

**Comment by jantonguirao** [comment]
!build

**Comment by jantonguirao** [comment]
!build

**Pull request merged**